### PR TITLE
[4.0][com_joomlaupdate] convert to prepared statement

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes

use prepared statement for SQL

### Testing Instructions
test com_joomlaupdate

### Expected result
work as before

### Actual result
N/A

### CODE REVIEW

File: administrator/components/com_joomlaupdate/Model/UpdateModel.php [hunk 1]
```diff
@@ -94,15 +95,18 @@ public function applyUpdateSite()
 				$updateURL = 'https://update.joomla.org/core/list.xml';
 		}
 
+		$id = (int) ExtensionHelper::getExtensionRecord('files_joomla')->extension_id;
```
Comment (Quy) [inline administrator/components/com_joomlaupdate/Model/UpdateModel.php]:
```suggestion
		$id = ExtensionHelper::getExtensionRecord('files_joomla')->extension_id;
```

File: administrator/components/com_joomlaupdate/Model/UpdateModel.php [hunk 2]
```diff
@@ -1356,22 +1363,24 @@ public function getNonCoreExtensions()
...
+				$db->quoteName('ex.name'),
+				$db->quoteName('ex.extension_id'),
+				$db->quoteName('ex.manifest_cache'),
+				$db->quoteName('ex.type'),
+				$db->quoteName('ex.folder'),
+				$db->quoteName('ex.element'),
+				$db->quoteName('ex.client_id'),
+				$db->quoteName('si.location'),
+			]
 		)->from(
```
Comment (Quy) [inline administrator/components/com_joomlaupdate/Model/UpdateModel.php]: Please reformat this section to start `->` .

Comment (alikon) [reply administrator/components/com_joomlaupdate/Model/UpdateModel.php]: done

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 9bada55d95b28037d88f3ae1b720f463b27a263b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27085">issues.joomla.org/tracker/joomla-cms/27085</a>.</sub>

File: administrator/components/com_joomlaupdate/Model/UpdateModel.php [hunk 3]
```diff
@@ -94,15 +95,18 @@ public function applyUpdateSite()
...
 		$query = $db->getQuery(true)
 			->select($db->quoteName('us') . '.*')
-			->from($db->quoteName('#__update_sites_extensions') . ' AS ' . $db->quoteName('map'))
+			->from($db->quoteName('#__update_sites_extensions', 'map'))
 			->join(
-				'INNER', $db->quoteName('#__update_sites') . ' AS ' . $db->quoteName('us')
-				. ' ON (us.update_site_id = map.update_site_id)'
+				'INNER',
+				$db->quoteName('#__update_sites', 'us'),
+				$db->quoteName('us.update_site_id') . ' = ' . $db->quoteName('map.update_site_id'),
```
Comment (SharkyKZ) [inline administrator/components/com_joomlaupdate/Model/UpdateModel.php]:
```suggestion
				$db->quoteName('us.update_site_id') . ' = ' . $db->quoteName('map.update_site_id')
```
Trailing commas in function calls need PHP 7.3+.

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on b8d397e850afd39f3980bf40e68b2338a75e1516<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27085">issues.joomla.org/tracker/joomla-cms/27085</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27085">issues.joomla.org/tracker/joomla-cms/27085</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**